internal/charmstore/resources.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmstore // import "gopkg.in/juju/charmstore.v5-unstable/internal/charmstore"
+
+import (
+	"io"
+	"time"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+
+	"gopkg.in/juju/charmstore.v5-unstable/internal/mongodoc"
+)
+
+var resourceNotFound = errgo.Newf("resource not found")
+
+// ListResources returns the list of resources for the charm at the
+// latest revision for each resource.
+func (s Store) ListResources(entity *mongodoc.Entity, channel params.Channel) ([]*mongodoc.Resource, error) {
+	if entity.URL.Series == "bundle" {
+		return nil, errgo.Newf("bundles do not have resources")
```

Comment:
check with roger et. al. on this, but I think he mentioned that we should just return nothing for bundles, not an error.